<launch>	
<!-- CURRENTLY COMMENTED OUT: Makes viewing node for showing detected pedestrians
	<node name="image_view_SEMANTICSEGMENTATION" pkg="image_view" type="image_view" respawn="false" output="screen">
			<remap from="image" to="/semantic_segmentation/imageSS"/>
			<param name="autosize" value="false" />
	</node>
-->
<!-- CURRENTLY COMMENTED OUT: Makes viewing node for showing camera input 
	<node name="image_view_WEBCAM" pkg="image_view" type="image_view" respawn="false" output="screen">
			<remap from="image" to="/Logitech_webcam/image_raw"/>
			<param name="autosize" value="false" />
	</node>
-->

	<!-- Uncompressing logitech cam -->
	<node name="republish" pkg="image_transport" type="republish" args="compressed in:=/Logitech_webcam/image_raw _image_transport:=compressed raw out:=/Logitech_webcam/image_raw" respawn="false" output="screen">
	</node> 

	<!-- Semantic Segmentation node  -->
	<node type="semantic_segmentation.py" name="semantic_segmentation" pkg="semantic_segmentation" respawn="true" output="screen">
		<param name="topicInName" value="/Logitech_webcam/image_raw" />
		<param name="objectType_unknown" value="True" />
		<param name="objectType_grass" value="True" />
		<param name="objectType_ground" value="True" />
		<param name="objectType_human" value="True" />
		<param name="objectType_shelterbelt" value="True" />
		<param name="objectType_vehicle" value="True" />
		<param name="objectType_water" value="True" />
		<param name="imgDimWidth" value="600" />
		<param name="imgDimHeight" value="300" />
		<param name="gpuDevice" value="0" /> <!-- Select GPU device, negative values selects cpu -->
		<param name="dirModelDescription" value="$(find semantic_segmentation)/models/fcn-8s-pascal-deploy.prototxt" />
		<param name="dirModelVaules" value="$(find semantic_segmentation)/models/fcn-8s-pascalcontext.caffemodel" />
		<param name="dirTestImage" value="$(find semantic_segmentation)/src/Street2.jpg" />
		<param name="dirRemapping" value="$(find semantic_segmentation)/remappingObjectTypes.mat" />
	</node>


<!-- CURRENTLY COMMENTED OUT: Showing image detection for each object class 
	<node name="image_view0" pkg="image_view" type="image_view" respawn="false" output="screen">
			<remap from="image" to="/det/Logitech_webcam/semantic_segmentation/grass"/>
			<param name="autosize" value="false" />
			<param name="window_name" value="grass" />
	</node>
	<node name="image_view1" pkg="image_view" type="image_view" respawn="false" output="screen">
			<remap from="image" to="/det/Logitech_webcam/semantic_segmentation/ground"/>
			<param name="autosize" value="false" />
			<param name="window_name" value="ground" />
	</node>
	<node name="image_view2" pkg="image_view" type="image_view" respawn="false" output="screen">
			<remap from="image" to="/det/Logitech_webcam/semantic_segmentation/human"/>
			<param name="autosize" value="false" />
			<param name="window_name" value="human" />
	</node>
	<node name="image_view3" pkg="image_view" type="image_view" respawn="false" output="screen">
			<remap from="image" to="/det/Logitech_webcam/semantic_segmentation/shelterbelt"/>
			<param name="autosize" value="false" />
			<param name="window_name" value="shelterbelt" />
	</node>
	<node name="image_view4" pkg="image_view" type="image_view" respawn="false" output="screen">
			<remap from="image" to="/det/Logitech_webcam/semantic_segmentation/unknown"/>
			<param name="autosize" value="false" />
			<param name="window_name" value="unknown" />
	</node>
	<node name="image_view5" pkg="image_view" type="image_view" respawn="false" output="screen">
			<remap from="image" to="/det/Logitech_webcam/semantic_segmentation/vehicle"/>
			<param name="autosize" value="false" />
			<param name="window_name" value="vehicle" />
	</node>
	<node name="image_view6" pkg="image_view" type="image_view" respawn="false" output="screen">
			<remap from="image" to="/det/Logitech_webcam/semantic_segmentation/water"/>
			<param name="autosize" value="false" />
			<param name="window_name" value="water" />
	</node>
-->
</launch>
